core/src/main/java/io/questdb/griffin/engine/functions/str/CharIndexFunctionFactory.java

```diff
@@ -0,0 +1,134 @@
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.IntFunction;
+import io.questdb.griffin.engine.functions.constants.IntConstant;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+import org.jetbrains.annotations.NotNull;
+
+public class CharIndexFunctionFactory implements FunctionFactory {
+
+    private static final IntConstant NOT_FOUND = new IntConstant(0);
+
+    @Override
+    public String getSignature() {
+        return "charindex(SSI)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        final Function substrFunc = args.getQuick(0);
+        if (substrFunc.isConstant()) {
+            if (substrFunc.getStrLen(null) < 1) {
+                return NOT_FOUND;
+            }
+        }
+
+        final Function strFunc = args.getQuick(1);
+        if (strFunc.isConstant()) {
+            if (strFunc.getStrLen(null) < 1) {
+                return NOT_FOUND;
+            }
+        }
+
+        final Function startFunc = args.getQuick(2);
+
+        return new Func(substrFunc, strFunc, startFunc);
+    }
+
+    public static class Func extends IntFunction {
+
+        private final Function substrFunc;
+        private final Function strFunc;
+        private final Function startFunc; // positions start with 1
+
+        public Func(Function substrFunc, Function strFunc, Function startFunc) {
+            this.substrFunc = substrFunc;
+            this.strFunc = strFunc;
+            this.startFunc = startFunc;
+        }
+
+        @Override
+        public int getInt(Record rec) {
+            final CharSequence substr = this.substrFunc.getStr(rec);
+            final CharSequence str = this.strFunc.getStr(rec);
+            if (substr != null && str != null) {
+                final int start = this.startFunc.getInt(rec);
+                if (str.length() < start) {
+                    return 0;
+                }
+                return charIndex(substr, str, start);
+            }
+            return 0;
+        }
+
+        private int charIndex(@NotNull CharSequence substr, @NotNull CharSequence str, int start) {
+            final int substrLen = substr.length();
+            if (substrLen < 1) {
+                return 0;
+            }
+            final int strLen = str.length();
+            if (strLen < 1) {
+                return 0;
```

Comment:
Search for empty with `strpos` string returns 1 in postgres

Reply:
Fixed in 3bd6d763f242b6cf722d2a6418e7287eda9d60fe